crates/task-impls/src/consensus.rs

```diff
@@ -473,6 +478,23 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     }
                 }
 
+                // If we have a ViewSyncFinalize cert, only vote if it is valid.
+                if let Some(cert) = &self.view_sync_cert {
+                    // Do not vote for the proposal if the cert is valid for an incorrect view
+                    if cert.view_number != view {
+                        debug!(
+                            "Cert view number {:?} does not match proposal view number {:?}",
+                            cert.view_number, view
+                        );
+                        return;
+                    }
+
+                    if !cert.is_valid_cert(self.quorum_membership.as_ref()) {
+                        debug!("Invalid ViewSyncFinalize cert provided");
+                        return;
+                    }
+                }
```

Comment:
Hmm I don't see how the vote tests are passing with this.  This code is only reachable in the case where `proposal.data.justify_qc.get_view_number() == view - 1` but in that case were would not have been in view sync.  This logic needs work alongside the timeout logic.  

The logic needs to be if `proposal.data.justify_qc.get_view_number() == view - 1`  we're good else:  
- check if we have a timeout cert of view sync cert for the right view